### PR TITLE
API created, new features and code refactoring

### PR DESCRIPTION
**Pull request by ghost**
API created using json-server and blog updated with new features : add, delete and edit article;
Refactoring the code and comments added

### CODE REVIEW

File: blog-app/client/js/script.js [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+
+//FUNCTIONS
+// get articles from db.json
+function getArticles(uri) {
+
+    fetch('http://localhost:3000/articles')
+    .then(
+        function(response) {
+        if (response.status !== 200) {
+            console.log('Looks like there was a problem. Status Code: ' +
```
Comment (ghost) [inline blog-app/client/js/script.js]: nice idea! additionaly, you could show a notification to the user or an error page.

File: blog-app/client/js/script.js [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+
+        // Examine the text in the response
+        response.json().then(function(data) {
+            articles = data;
+            // call the displayHtml function to create the elements for the page based on the uri
+            displayHtml(uri);
+            if(uri === "home") {
+
+                // CONSTANTS
+                const ADDBTN = document.getElementById("add-article");
```
Comment (ghost) [inline blog-app/client/js/script.js]: you could move the code for the modal inside the displayHtml function, where you check for the path "home"

File: blog-app/client/js/script.js [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+    let title = document.createElement("h2");
+    title.setAttribute("class","title");
+    title.textContent = article.title;
+
+    let ul = document.createElement("ul");
+    ul.setAttribute("class","info__container");
+    for(let i in article.info){
+        let li = document.createElement("li");
+        li.setAttribute("class","info__item");
+        if(i === "author") {
```
Comment (ghost) [inline blog-app/client/js/script.js:440]: well thought json structure!

File: blog-app/client/.vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "liveServer.settings.port": 5501
```
Comment (alexandrumalinescu) [inline blog-app/client/.vscode/settings.json:2]: better go in port 3001

**Review by ghost — APPROVED** [review]
- well put comments that explain the big steps in your code 
- easy to follow functions

File: blog-app/client/js/script.js (hunk 3 above)
Comment (ghost) [reply blog-app/client/js/script.js:440]: Thank you!

File: blog-app/client/js/script.js (hunk 2 above)
Comment (ghost) [reply blog-app/client/js/script.js]: I updated the code.

File: blog-app/client/js/script.js (hunk 1 above)
Comment (ghost) [reply blog-app/client/js/script.js]: I will implement this at a later time. Thank you for the idea.

**Pull request merged**